plugins/bbs_exporter.js

```diff
@@ -0,0 +1,368 @@
+
+         return output;
+    }
+
+    function compileFirstCubes()
+    {
+        var createCubes;
+
+        createCubes = function (objects)
+        {
+            var groups = [];
+            var cubes = [];
+
+            for (let i = 0; i < objects.length; i++)
+            {
+                var object = objects[i];
+
+                if (object.type === "group")
+                {
+                    groups.push(object);
+                }
+                else if (object.type === "cube")
+                {
+                    cubes.push(object);
+                }
+            }
+
+            if (cubes.length > 0)
+            {
+                var result = [];
+
+                for (let i = 0; i < cubes.length; i++)
+                {
+                    result.push(createCube(cubes[i]));
+                }
+
+                return result;
+            }
+
+            for (let i = 0; i < groups.length; i++)
+            {
+                var result = createCubes(groups[i].children);
+
+                if (result)
+                {
+                    return result;
+                }
+            }
+
+            return false;
+        }
+
+        return createCubes(Outliner.root);
+    }
+
+    /* Bootstrap */
+
+    var bbsCodec = new Codec("bbs_model", {
+        name: "BBS model",
+        extension: "json",
+        remember: false,
+        compile(options) {
+            return autoStringify(compile());
+        }
+    });
+
+    var exportDialog = new Dialog({
+        id: "bbs_export_options",
+        title: "BBS model exporter",
+        form: {
+            exportModel: {
+                label: "Export model data",
+                type: "checkbox",
+                value: true
+            },
+            exportAnimations: {
+                label: "Export animations",
+                type: "checkbox",
+                value: true
+            },
+            copyToBuffer: {
+                label: "Copy to buffer",
+                type: "checkbox",
+                value: false
+            },
+            copyOnlyFirst: {
+                label: "Copy first group",
+                description: "When enabled, copies to the buffer only cubes from the first found group. This option is ignored when Copy to buffer option is disabled!",
+                type: "checkbox",
+                value: false
+            }
+        },
+        onConfirm: function(formData) {
+            this.hide();
+
+            lastOptions.model = formData.exportModel;
+            lastOptions.animations = formData.exportAnimations;
+
+            if (formData.copyToBuffer)
+            {
+                var data = {};
+
+                if (formData.copyOnlyFirst)
+                {
+                    data = compileFirstCubes();
+                }
+                else
+                {
+                    data = compile();
+                }
+
+                electron.clipboard.writeText(autoStringify(data));
```

Comment:
Use `Clipbench.setText(text)` to write to the clipboard, instead of requiring Electron. That way it will also work in the web app.

Reply:
Oh, I didn't know that, thanks!